PEP 1: Update main PEP discussion venue from Python-Dev to Discourse (#2775)
python/peps: pep-0001.txt

* A direct link to the current discussion thread is provided in the PEP
  under the ``Discussions-To`` header.

Typically, the `PEPs category`_ of the `Python Discourse`_

> is a good choice for most new PEPs,
> whereas historically the `Python-Dev`_ mailing list was commonly used.

Reviewer: PEP 1 is very long, I'd suggest cutting the second part of this sentence for brevity
```suggestion
is a good choice for most new PEPs.
```

Author: I certainly agree that PEP 1 is rather long, and could do with being more shorter and focused at the macro level (and have some much more impactful ideas for this). That said, this is the micro level of less than a dozen words out of 6500; whether we cut out a relatively concise mention is more a question of whether it serves a purpose consummate with its fairly brief length.

In this case, I retained this line, the only reference left to Python-Dev in the PEP, since I felt it provided important context for historical PEPs, a large majority of which have their discussions and resolutions on that venue, many without one or both being explicitly mentioned/linked in the PEP in question.

However, if others feel that doesn't justify even the relatively brief mention, I can elide it as you suggest.

Some specialized topics have specific venues, such as
`Typing-SIG`_ for typing PEPs or the `Packaging category`_ on the Python
Discourse for packaging PEPs. If the PEP authors are unsure of the best venue,
the PEP Sponsor and PEP editors can advise them accordingly.

> If the chosen venue is not a Discourse category,

Author: This is actually a key point that I meant to bring up, thanks. @encukou , any guidance on this?

Historically, in the special case of PEPs that had, per SC permission, their canonical discussion location in another Discourse category (i.e. Packaging), at least recently they've generally not been also posted to Python-Dev (and never the PEPs category), and to require this seemed somewhat noisy, redundant and carries the risk of splitting the discussion between different threads on the very same platform (without Discourse admin intervention to immediately lock the thread each time). Additionally, those just looking for notifications of new PEPs being posted can subscribe to the PEP RSS feed, etc.

However, if @encukou and the SC would like this worded differently, or there's otherwise PEP editor consensus, then I'm of course happy to oblige.

> If the chosen venue is not a Discourse category,

Author: Thanks; I ended up rewriting this to consolidate these two mentions of posting on the PEPs category if the discussion thread is not there into one, which made it many lines shorter, as well as being specific about it being the PEPs category as requested.

a brief announcement thread should be posted in ``PEPs``
when the draft PEP is committed to the repository,
which should include a link to the rendered PEP
and to the canonical ``Discussions-To`` thread.

If a PEP undergoes a significant re-write or other major, substantive
changes to its proposed specification, a new thread should typically be created
in the chosen venue to solicit additional feedback. If this occurs, the
``Discussions-To`` link must be updated and a new ``Post-History`` entry added

> pointing to this new thread, and (if Discourse is not the discussion venue)
> a further announcement thread should generally posted
> in the ``PEPs`` Discourse category
> containing the same information as above

Reviewer: Shortened for brevity
```suggestion
pointing to this new thread, and (if not the discussion venue) a further
announcement should be posted in the `PEPs Discourse category`_
containing the same information as above
```

Author: I (manually) applied most of the suggested elisions (except for the change pending discussion above) while fixing the breaks and the link target name, and I also split it into two sentences to make it easier to read and follow.

Author: I ended up making things many lines shorter by consolidating the two mentions of posting an announcement thread on the PEPs category into one, as well as making the language more concise; also, I implemented the suggestion to explicitly specify the PEPs category per the above

the Steering Council.  This logic is intentionally circular.)  See :pep:`2`
for standard library module acceptance criteria.

Except where otherwise approved by the Steering Council,
pronouncements of PEP resolution will be posted to the
`PEPs category`_ on the `Python Discourse`_

> and the PEP's most recent ``Discussions-To`` thread.

Author: Applied manually to fix the link target names

When a PEP is Accepted, Rejected or Withdrawn, the PEP should be updated
accordingly. In addition to updating the Status field, at the very least
the Resolution header should be added with a direct link

> to the relevant SC post officially accepting or rejecting the PEP.

Author: IMO, I don't believe cutting two words out of 6500 (0.03%) is worth the clarity loss here in specifically referring to the official SC acceptance, as opposed to another plausible target, but what do others think?

> to the relevant SC post officially accepting or rejecting the PEP.

Author: I was thinking potentially a post by e.g. the PEP delegate stating informally that they are going to accept the PEP prior to an official announcement, or were very likely to, or would pending some change—I ran into that a fair bit researching older PEP discussion threads, but it seem fairly formalized nowadays, so my concerns were probably overblown. Also, to your point, not only does the SC wording technically exclude a PEP delegate, but also, to wit, "accepting or rejecting" the PEP doesn't include Withdrawn, so I'll revise this to say

```suggestion
to the relevant post making a decision on the PEP.
```

which is both shorter and more accurate
